doc/whats-new.rst

```diff
@@ -36,7 +36,10 @@ Breaking changes
 
 Bug fixes
 ~~~~~~~~~
-
+- Fix for :py:meth:`DataSet.diff` and consequently :py:meth:`DataArray.diff` to
+  raise a KeyError when `diff` is parameterized with a dimension that does not
+  exist in callee's DataSet/DataArray. (:issue:`7748`)
```

Comment:
This is fine, though we can also simplify the language a bit — cut words like `consequently` / `parameterized with`!